### ISSUE TITLE
Disconnect/infinite loop when loading profile pages

### ISSUE THREAD

**Issue by SnugglePilot** [nguyenquyhy/Flight-Tracker-StreamDeck#24]
Sometimes when I load a profile page (eg: swap to my engine readout stack) and don't hit any buttons, flight sim will enter an infinite loop and streamdeck needs to be closed for several minutes, then re-opened.

I think this is because of all of the polling of current state information to populate button states - the simconnect interface gets overwhelemd? Guessing here.

I'm wondering, if that is the problem, if it is possible to "queue" requests for information and trickle them out at a metered rate? Hopefully this will prevent the crashes in the future.

**Comment by nguyenquyhy** [comment]
About the "Infinite loop", are you seeing the Connected to Stream Deck message flashing constantly?

**Comment by SnugglePilot** [comment]
Yep, that's what I meant by that. (This is without calling any Events into the sytem.)

**Comment by nguyenquyhy** [comment]
That is actually from one of my workaround for a SimConnect problem I encountered during Alpha: when I clicked a button too many times too quickly, SimConnect returned a generic error (without any details), and stop sending data to the plugin without terminating the connection.

My solution was to terminate the connection and reconnect. I seem work well for my case.

However, later I realized that there was another issue from SimConnect that also return that generic error, but cannot be solved by an immediate reconnection. This led to an "infinite" disconnect-reconnect attempt from the plugin (because SimConnect immediately send that generic error after the reconnection). As you said, disconnect and wait for a while for SimConnect to recover seem to fix this issue.

I am thinking of 1 of the 2 ways to solve this issue:
- Remove my disconnect-reconnect logic and see if the original issue is still there. Users might get weird period when the stream deck looks fine but actually doesn't receive any data from SimConnect though.
- Implementing some throttling for the reconnection, so if it has reconnected a couple times recently, it will wait for a couple more seconds.
